base/utf8.jl

```diff
@@ -112,8 +103,11 @@ function getindex(s::UTF8String, r::UnitRange{Int})
     isempty(r) && return empty_utf8
     i, j = first(r), last(r)
     d = s.data
+    if i < 1 || i > length(s.data)
```

Comment:
Ah, got it, you count on `is_utf8_start(d[i])` to catch other cases below. As you like.